usr/share/rear/lib/_input-output-functions.sh

```diff
@@ -993,8 +993,173 @@ function LogPrintIfError () {
+#   REAR_DIR_PREFIX=/rear/prefix
+#   SHARE_DIR=/rear/prefix/usr/share/rear
+#   CONFIG_DIR=/rear/prefix/etc/rear
+#   VAR_DIR=/rear/prefix/var/lib/rear
+#   LOG_DIR=/rear/prefix/var/log/rear
+# (B2)
+# Special case (i.e. Git checkout on the original system) in the ReaR recovery system:
+# ReaR basic directories when running /bin/rear in the ReaR recovery system
+# when ReaR on the original system was a Git checkout:
+#   REAR_DIR_PREFIX=
+#   SHARE_DIR=/usr/share/rear
+#   CONFIG_DIR=/etc/rear
+#   VAR_DIR=/var/lib/rear
+#   LOG_DIR=/var/log/rear
+# BUT some of them are symlinks and others are not:
+#   /usr/share/rear is a symbolic link to ../../rear/prefix/usr/share/rear
+#   /etc/rear is a normal directory (no symbolic link to /rear/prefix/etc/rear)
+#   /var/lib/rear is a symbolic link to ../../rear/prefix/var/lib/rear
+#   /var/log/rear is a normal directory (no symbolic link to /rear/prefix/var/log/rear)
+# so the actual files in /usr/share/rear and /var/lib/rear are under a prefix directory
+# except the actual files in /etc/rear and /var/log/rear which are really in /etc/rear and /var/log/rear
+# so the empty REAR_DIR_PREFIX matches where the actual files in CONFIG_DIR and LOG_DIR are
+# but it does neither match where the actual files in SHARE_DIR are
+# nor does REAR_DIR_PREFIX match where the actual files in VAR_DIR are.
+# Cf. https://github.com/rear/rear/pull/3379#issuecomment-2598270427
+# and https://github.com/rear/rear/pull/3379#issuecomment-2601575741
+# SHARE_DIR is where ReaR's scripts are so it is most often used and therefore it should be first
+# in TRUSTWORTHY_PATHS to return early from the 'for trustworthy_path ...' testing loop in is_trustworthy_path().
+# CONFIG_DIR is used to source e.g. etc/rear/local.conf
+# VAR_DIR is used by "rear recover" to source /var/lib/rear/layout/diskrestore.sh in 200_run_layout_code.sh
+# According to https://github.com/rear/rear/issues/3259#issuecomment-2385745545
+# '/usr/' '/etc/' '/lib/' should be considered in general to be trustworthy paths
+# so when SHARE_DIR is a sub-directory of '/usr/' it is needless to also add SHARE_DIR
+# and when CONFIG_DIR is a sub-directory of '/etc/' it is needless to also add CONFIG_DIR.
+# The trailing / ensures that e.g. /libanywhere/ is not falsely regarded as trustworthy path
+# so in particular /lib64/ is currently not considered as trustworthy path.
+if test ${#TRUSTWORTHY_PATHS[@]} -eq 0 ; then
+    # Set default when TRUSTWORTHY_PATHS is empty:
+    if test "$REAR_DIR_PREFIX" ; then
+        # When REAR_DIR_PREFIX is set it is case (B1) i.e. Git checkout on the original system.
+        # All files below a Git checkout directory are considered to be trustworthy
+        # see https://relax-and-recover.org/documentation/security-architecture
+        # SHARE_DIR and CONFIG_DIR and VAR_DIR are sub-directories of REAR_DIR_PREFIX:
+        TRUSTWORTHY_PATHS=( "$REAR_DIR_PREFIX/" '/usr/' '/etc/' '/lib/' )
```

Comment:
Hmm, why these system dirs? What about `/run` or others where we might want to source files from?

Reply:
Because @pcahyna proposed them in
https://github.com/rear/rear/issues/3259#issuecomment-2385745545